src/pykiso/test_coordinator/test_suite.py

```diff
@@ -364,6 +371,69 @@ def __init__(
         # add sorted test case list to test suite
         self.addTests(test_case_list)
 
+        self.failed_suite_setups = set()
+
+    def check_suite_setup_failed(self, test: BasicTest, result: TestResult) -> None:
+        """Check if the suite setup has failed and store failed suite id.
+
+        :param test: test to check
+        :param result: unittest result object
+        """
+        if isinstance(test, BasicTestSuiteSetup):
+            for suite_type, _ in result.failures:
+                if isinstance(suite_type, BasicTestSuiteSetup):
+                    self.failed_suite_setups.add(test.test_suite_id)
+
+    def run(self, result: TestResult, debug: bool = False) -> TestResult:
+        """Override run method from unittest.suite.TestSuite.
+        Added functionality:
+        Skip suite tests if the parent test suite setup has failed.
+
+        :param result: unittest result storage
+        :param debug: True to enter debug mode, defaults to False
+        :return: test suite result
+        """
+        topLevel = False
+        if getattr(result, "_testRunEntered", False) is False:
+            result._testRunEntered = topLevel = True
+
+        for index, test in enumerate(self):
+            if result.shouldStop:
+                break
+
+            if _isnotsuite(test):
+                self._tearDownPreviousClass(test, result)
+                self._handleModuleFixture(test, result)
+                self._handleClassSetUp(test, result)
+                result._previousTestClass = test.__class__
+
+                if getattr(test.__class__, "_classSetupFailed", False) or getattr(
+                    result, "_moduleSetUpFailed", False
+                ):
+                    continue
+
+            if not debug:
+                if test.test_suite_id in self.failed_suite_setups:
+                    result.addSkip(
+                        test,
+                        f"Suite Setup failed for test suite {test.test_suite_id} failed",
```

Comment:
```suggestion
                        f"Suite Setup failed for test suite {test.test_suite_id}",
```

Reply:
change to one line 
if getattr(test, "_testMethodDoc", None) is not None